src/BioSequences.jl

```diff
@@ -227,4 +227,46 @@ include("search/ApproxSearchQuery.jl")
 include("search/re.jl")
 include("search/pwm.jl")
 
+
+
+struct Search{Q,I}
+    query::Q
+    itr::I
+    start::Integer
+    stop::Integer
+    overlap::Bool
+end
+
+const default_overlap = true
+
+search(query, itr, start=firstindex(itr); overlap = default_overlap, stop = lastindex(itr)) = Search(query, itr, start, stop, overlap)
+
+function Base.iterate(itr::Search, state=itr.start)
+    val = findnext(itr.query, itr.itr, state)
+    val === nothing && return nothing
+    last(val) > itr.stop && return nothing
+    state = itr.overlap ? first(val) + 1 : last(val) + 1
+    return val, state
+end
+
+const HasRangeEltype = Union{<:ExactSearchQuery, <:ApproximateSearchQuery, <:Regex}
+
+Base.eltype(::Type{Search{Q,I}}) where {Q<:HasRangeEltype,I} = UnitRange{Int}
+Base.IteratorEltype(::Type{Search{Q,I}}) where {Q<:HasRangeEltype,I} = Base.IteratorEltype(UnitRange{Int})
+
+Base.eltype(::Type{<:Search}) = Int
+Base.IteratorEltype(::Type{<:Search}) = Base.IteratorEltype(Int)
+
+Base.IteratorSize(::Type{<:Search}) = Base.SizeUnknown()
+
+function Base.findall(pat, seq::BioSequence, start::Integer=firstindex(seq); overlap::Bool = default_overlap, stop::Integer = lastindex(seq))
```

Comment:
I'm ambivalent about this signature. Why is start part of the signature but stop isn't? Might be good though, just not sure.

Reply:
I think I would prefer having the same signature as `Base.findall(::AbstractString, ::AbstractString; overlap=true)` - though `Base` has overlap default to `false`, that seems strange to me.

If the user wants to limit the start and stop, they can create a view of the sequence.

Reply:
An overlap default of `true` makes more sense to me in the context of BioSequences. But I'm okay changing to be consistent with `Base` if that's the consensus. However, setting the default to `false` may introduce a risk of missing important sequences.

Reply:
I agree - `true` is the most sensible default here 👍